### ISSUE TITLE
WIFI TimeSync

### ISSUE THREAD

**Issue by sajjadakhter** [espressif/esp-idf#4922]
is there any chip that supports Wifi TimeSync
Similar question was asked almost 2 years ago, i thought to check if any thing change 
even if its planned for future.

**Comment by sajjadakhter** [comment]
https://github.com/espressif/esp-idf/issues/2326

**Comment by liuzfesp** [comment]
Hi @sajjadakhter I think ESP32 S2 chip supports it, but the software is not ready yet. Please correct me if I'm wrong @sagb2015 .

**Comment by sagb2015** [comment]
@sajjadakhter Yes ESP32 S2 chip support timesync. We are in process of implementing FTM feature as per IEEE specification.

**Comment by sajjadakhter** [comment]
> @sajjadakhter Yes ESP32 S2 chip support timesync. We are in process of implementing FTM feature as per IEEE specification.

Any timeline? will it be able to sync over esp-now as well or only wifi?

**Comment by dinuchiriac** [comment]
I'm interested in this as well. Do we have an estimated timeline when will be able to test TimeSync?

**Comment by sagb2015** [comment]
@dinuchiriac We are currently targeting the availability by Mid July.

**Comment by dinuchiriacsonos** [comment]
@sagb2015  Thanks,   Looking forward to testing it.

**Comment by reletobg** [comment]
Is there any updates around the TimeSync? @sagb2015

**Comment by sagb2015** [comment]
@reletobg  This feature is under test. However, we need more time to evaluate accuracy under different environmental conditions.

**Comment by sagb2015** [comment]
Beta release for the feature is [here](https://github.com/espressif/esp-idf/tree/feature/ftm_support) . The example can be found [here](https://github.com/espressif/esp-idf/tree/feature/ftm_support/examples/wifi/ftm).

**Comment by sajjadakhter** [comment]
Is there an example to use FTM for time sync. Trying to save myself some coding if its already done.

**Comment by sagb2015** [comment]
@sajjadakhter  I need to apologise on this. Actually,  the Timesync program from WFA and FTM (WiFi location) are two different things even though FTM hardware can be used for Timesync (T1-T4) values to be precise.  We will look into this feature request.

**Comment by VladimirP1** [comment]
Any update on this?
I would be interested in accurate time sync over wifi

**Comment by nachiketkukade** [comment]
@sajjadakhter , WiFi TimeSync program doesn't exist anymore OR delayed. As of now - except 802.1AS-2020 over Wi-Fi - there is no standard way to perform Wi-Fi based time synchronisation. We do not support 802.1AS, however it's possible to perform Time Synchronisation between 2 ESP devices using FTM. 
@VladimirP1 , if interested we can provide patches that can work as PoC for FTM based Time Synchronisation

**Comment by nachiketkukade** [comment]
No immediate plan to support 802.1AS over WiFi, closing
